File: mywebsite-parent/mywebsite-content-interface/src/main/java/cn/boom/mywebsite/content/service/UserService.java
package cn.boom.mywebsite.content.service;

import cn.boom.mywebsite.pojo.MyWebSiteResult;
import cn.boom.mywebsite.pojo.TbUser;

import java.util.List;

public interface UserService {

    /**
     * 查询所有
     * @return
     */
    public abstract List<TbUser> findAll();
    /**
     * 根据Id查询
     * @param id
     * @return
     */
    public abstract TbUser findOneById(Long id);

    /**
     * 根据邮箱查询
     * @param email
     * @return
     */
    public abstract TbUser findOneByEmail(String email);

    /**
     * 条件分页搜索
     * @param user
     * @param pageNum
     * @param pageSize
     * @return
     */
    public abstract MyWebSiteResult search(TbUser user, int pageNum, int pageSize);

    /**
     * 添加用户
     * @param user
     * @return
     */
    public abstract MyWebSiteResult add(TbUser user);

    /**
     * 修改用户
     * @param user
     * @return
     */
    public abstract MyWebSiteResult update(TbUser user);

    /**
     * 删除用户
     * @param ids
     * @return
     */
    public abstract MyWebSiteResult deleteIds(Long[] ids);
}

File: mywebsite-parent-v2.0/mywebsite-manager-web/src/main/webapp/angularjs/js/service/userService.js
app.service('userService', function ($http) {


    /**
     * 查询所有用户
     * @returns {*}
     */
    this.findAll = function () {
        return $http.get('../user/findAll');
    };

    /**
     * 条件分页搜索
     * @param searchEntity
     * @param page
     * @param rows
     * @returns {*}
     */
    this.search = function (searchEntity, page, rows) {
        return $http.post('../user/search?page=' + page + '&rows=' + rows, searchEntity);
    };

    /**
     * 根据id查询用户
     * @param id
     * @returns {*}
     */
    this.findOneById = function (id) {
        return $http.get('../user/findOneById?id=' + id);
    };

    /**
     * 更新用户
     * @param entity
     * @returns {*}
     */
    this.update = function (entity) {
        return $http.post('../user/update', entity);
    };


    /**
     * 添加用户
     * @param entity
     * @returns {*}
     */
    this.add = function (entity) {
        return $http.post('../user/add', entity);
    };

    /**
     * 批量删除
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../user/deleteIds?ids='+ids);
    };
});